core/src/main/scala/caliban/Value.scala

```diff
@@ -13,6 +13,9 @@ sealed trait InputValue extends Serializable { self =>
   def toInputString: String = ValueRenderer.inputValueRenderer.renderCompact(self)
 }
 object InputValue {
+  case object UndefinedValue                              extends InputValue {
```

Comment:
As I mentioned we don't need that as we can use `buildMissing` in `ArgBuilder`

Reply:
What default value are you suggesting for a missing field? IMHO, the case of a missing field should be solved on the level of the domain-logic, not in deserializer.

Reply:
Domain logic does not use `InputValue` either. `ArgBuilder` allows transforming the input into whatever type is used in business logic.

Your `missingInput` function can be implemented without `UndefinedValue`:
```scala
def missingInput[A, F[_]](onMissing: F[A], provided: A => F[A])(implicit ev: ArgBuilder[A]): ArgBuilder[F[A]] =
  new ArgBuilder[F[A]] {
    override def buildMissing(default: Option[String]): Either[ExecutionError, F[A]] = Right(onMissing)
    def build(input: InputValue): Either[ExecutionError, F[A]]                       = ev.build(input).map(provided)
  }
```